akka-projection-kafka/src/main/scala/akka/projection/kafka/KafkaSourceProvider.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.kafka
+
+import akka.annotation.InternalApi
+import akka.kafka.ConsumerMessage.CommittableMessage
+import akka.kafka.ConsumerMessage.PartitionOffset
+import akka.kafka.ConsumerSettings
+import akka.kafka.Subscription
+import akka.kafka.scaladsl.Consumer
+import akka.projection.scaladsl.SourceProvider
+import akka.stream.scaladsl.Source
+import org.apache.kafka.common.TopicPartition
+
+object KafkaSourceProvider {
+
+  /**
+   * Create a [[SourceProvider]] that resumes from Kafka managed offsets
+   */
+  def fromConsumerGroup[Key, ValueEnvelope](
+      settings: ConsumerSettings[Key, ValueEnvelope],
+      subscription: Subscription): SourceProvider[PartitionOffset, CommittableMessage[Key, ValueEnvelope]] =
+    new KafkaOffsetSourceProvider[Key, ValueEnvelope](settings, subscription)
+
+  /**
+   * TODO
+   *
+   * Create a [[SourceProvider]] that resumes from externally managed offsets
+   *
+   * NOTE: Uses Alpakka Kafka Consumer.plainPartitionedManualOffsetSource that merges all partitions back into
+   * one stream
+   */
+  def fromOffsets[Key, ValueEnvelope](
```

Comment:
I'm not sure if this would be useful to implement, or not.

Reply:
Yes.  I've refactored the `SourceProvider` to support loading from external offsets.